packages/cli/src/TestWebhooks.ts

```diff
@@ -274,45 +284,42 @@ export class TestWebhooks implements IWebhookManager {
 		return foundWebhook;
 	}
 
-	async activateWebhook(
-		workflow: Workflow,
-		webhook: IWebhookData,
-		executionMode: WorkflowExecuteMode,
-		activationMode: WorkflowActivateMode,
-	) {
+	async activateWebhook(workflow: Workflow, webhook: IWebhookData) {
 		webhook.path = removeTrailingSlash(webhook.path);
+		webhook.isTest = true;
 
-		const key = this.toWebhookKey(webhook);
+		/**
+		 * Remove additional data from webhook because:
+		 *
+		 * - It is not needed for the test webhook to be executed.
+		 * - It contains circular refs that cannot be be cached.
+		 */
+		const { workflowExecuteAdditionalData: _, ...rest } = webhook;
 
-		webhook.isTest = true;
-		this.registrations[key].webhook = webhook;
```

Comment:
Does `key` already exist when this is happening? If so, can we already set the `webhook` property when it's created just to prevent another hop to the cache layer?

Reply:
Simplified this legacy stuff here: [5c2d670](https://github.com/n8n-io/n8n/pull/8176/commits/5c2d670e47d8a50c2c3bd7ad1271de8d6a6ba066)